Projects/Embark/Sources/Viewables/Actions/TextActionSet.swift

```diff
@@ -72,6 +98,36 @@ extension TextActionSet: Viewable {
                     }
                 }
             }
+            
+            bag += containerView.applyShadow({ (traitCollection) -> UIView.ShadowProperties in
+                UIView.ShadowProperties(
+                    opacity: 1,
+                    offset: .init(width: 0, height: 2),
+                    radius: 2,
+                    color: .brand(.secondaryShadowColor),
+                    path: nil,
+                    corners: (.allCorners, 8))
+            })
+            
+            let button = Button(
+                title: data.textActionSetData?.link.label ?? "",
+                type: .standard(backgroundColor: .black, textColor: .white),
+                isEnabled: false
+            )
+
+            bag += view.addArranged(button)
+            
+            func isValid(signal: ReadWriteSignal<String>, action: TextAction) -> Signal<Bool> {
+                signal.map { text in
+                    return text.count > 0
+                }.plain()
+            }
+            
+            bag += combineLatest(textActions!.map { signal, _, action in isValid(signal: signal, action: action) }).map {
```

Comment:
Avoid unwrapping here

Reply:
Maybe you could make this a bit cleaner with allSatisfy?

Reply:
That's interesting can you give me an example?